### PR TITLE
V2.4.14 openharmony bug fix

### PR DESCRIPTION
**Pull request by zhaoyang57**
1. Fix the issue of inaccurate calculation of sleep time when locking frames.
2. Fix the issue of UV timer not stopped when sliding up to the multitask list.
3. Specify the audio stream type as GAME.
4. Fix the issue of custom font fontCollection being incorrectly destroyed.
5. Adjust editbox component position

### CODE REVIEW

File: templates/js-template-link/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/components/EditBoxDialog.ets [hunk 1]
```diff
@@ -60,7 +60,7 @@ export struct EditBoxDialog {
         })
       }.padding({ left: 8, right: 8, top: 8, bottom: 8 })
       .backgroundColor(Color.Gray)
-
+      .offset({ x : 0, y : 15 })
```
Comment (qiuguohua) [inline templates/js-template-link/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/components/EditBoxDialog.ets:63]: What is the reason for doing this?

File: cocos/scripting/js-bindings/manual/jsb_platform_openharmony.cpp [hunk 2]
```diff
@@ -40,7 +40,9 @@
 
 using namespace cocos2d;
 
-static std::unordered_map<std::string, OH_Drawing_FontCollection*> _fontCollectionMap;
+static std::unordered_map<std::string, OH_Drawing_FontCollection*> _fontCollectionMap = {
+    {"openharmony_system_font", OH_Drawing_CreateSharedFontCollection()}
```
Comment (qiuguohua) [inline cocos/scripting/js-bindings/manual/jsb_platform_openharmony.cpp]: Can we use constant strings?

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp [hunk 3]
```diff
@@ -208,15 +208,24 @@ void OpenHarmonyPlatform::onCreateNative(napi_env env, uv_loop_t* loop) {
...
     LOGD("OpenHarmonyPlatform::onHideNative");
     EventDispatcher::dispatchOnPauseEvent();
+    if (_timerInited) {
+        uv_timer_stop(&_timerHandle);
+    }
 }
 
 void OpenHarmonyPlatform::onDestroyNative() {
     LOGD("OpenHarmonyPlatform::onDestroyNative");
+     if (_timerInited) {
```
Comment (qiuguohua) [inline cocos/platform/openharmony/OpenHarmonyPlatform.cpp:226]: Can we use _timerHandle!=null?

File: templates/js-template-link/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/components/EditBoxDialog.ets (hunk 1 above)
Comment (zhaoyang57) [reply templates/js-template-link/frameworks/runtime-src/proj.openharmony/entry/src/main/ets/components/EditBoxDialog.ets:63]: To optimize the component layout. 
If not set, there will be a significant gap between the layout of the text input box and the system input method.

File: cocos/platform/openharmony/OpenHarmonyPlatform.cpp (hunk 3 above)
Comment (zhaoyang57) [reply cocos/platform/openharmony/OpenHarmonyPlatform.cpp:226]: _timerHandle cannot be compared with null

File: cocos/scripting/js-bindings/manual/jsb_platform_openharmony.cpp (hunk 2 above)
Comment (zhaoyang57) [reply cocos/scripting/js-bindings/manual/jsb_platform_openharmony.cpp]: Fixed.

**Review by qiuguohua — APPROVED** [review]

**Comment by Sunny-guangge** [comment]
鸿蒙 restart有什么方案吗

**Comment by zhaoyang57** [comment]
> 鸿蒙 restart有什么方案吗

如果js引擎使用v8可以通过cc.game.restart()进行重启，使用方舟暂不支持

**Comment by Sunny-guangge** [comment]
鸿蒙Next可以使用V8引擎吗

**Comment by zhaoyang57** [comment]
> 鸿蒙Next可以使用V8引擎吗

2.4.14版本会支持

**Comment by Sunny-guangge** [comment]
我看现在2.4.14已经发布了   是这个版本吗

**Comment by Sunny-guangge** [comment]
> > 鸿蒙Next可以使用V8引擎吗
> 
> 2.4.14版本会支持

我看git记录里 2.4.14又把harmony v8删掉了

**Review by qiuguohua — APPROVED** [review]

**Pull request merged**